src/functions/retryProcessing/application/__tests__/AutoSaveOldUploadDeletion.int.ts

```diff
@@ -0,0 +1,167 @@
+      database: 'results',
+      password: 'Pa55word1',
+      port: 3306,
+    });
+    retryProcessor = new RetryProcessor(db);
+  });
+
+  describe('Clean up PROCESSED UPLOAD_QUEUE records', () => {
+    beforeEach(async () => {
+      const testResultData = getTestResultData();
+      const queueResultData = getQueueResultData();
+
+      await insertAutosaveTestResultData(db, testResultData);
+      await insertAutosaveQueueResultData(db, queueResultData);
+    });
+
+    afterEach(async () => {
+      await deleteAutosaveTestResultData(db, 'TEST_RESULT', [56, 57, 58, 59]);
+      await deleteAutosaveTestResultData(db, 'UPLOAD_QUEUE', [56, 57, 58, 59]);
+    });
+
+    it('should delete UPLOAD_QUEUE records that are PROCESSED with a timestamp less than the start_date provided',
+       async () => {
+         const changedRowCount = await retryProcessor.processOldEntryCleanup(1);
+         const autosaveQueueRecords = await getAutosaveQueueRecords(db);
+
+         expect(changedRowCount).toBe(6);
+
+        // assert UPLOAD_QUEUE records have been deleted for app-ref 56 and 57
+         expect(autosaveQueueRecords).not.toContain({ application_reference: 56, interface: 0, upload_status: 1 });
+         expect(autosaveQueueRecords).not.toContain({ application_reference: 56, interface: 1, upload_status: 1 });
+         expect(autosaveQueueRecords).not.toContain({ application_reference: 56, interface: 2, upload_status: 1 });
+         expect(autosaveQueueRecords).not.toContain({ application_reference: 57, interface: 0, upload_status: 1 });
+         expect(autosaveQueueRecords).not.toContain({ application_reference: 57, interface: 1, upload_status: 1 });
+         expect(autosaveQueueRecords).not.toContain({ application_reference: 57, interface: 2, upload_status: 1 });
+
+        // assert UPLOAD_QUEUE records have been retained for app-ref 58
+         expect(autosaveQueueRecords).toContain({ application_reference: 58, interface: 0, upload_status: 1 });
+         expect(autosaveQueueRecords).toContain({ application_reference: 58, interface: 1, upload_status: 1 });
+         expect(autosaveQueueRecords).toContain({ application_reference: 58, interface: 2, upload_status: 0 });
+       });
+
+  });
+
+  const getTestResultData = (): AutosaveTestData[] => {
+    return [
+      {
+        applicationReference: 56,
+        staffNumber: '1',
+        driverSurname: 'Bloggs',
+        resultStatus: 1, // ACCEPTED
+        autosave: true,
+      },
+      {
+        applicationReference: 57,
+        staffNumber: '1',
+        driverSurname: 'Bloggs',
+        resultStatus: 1, // ACCEPTED
+        autosave: true,
+      },
+      {
+        applicationReference: 58,
+        staffNumber: '1',
+        driverSurname: 'Bloggs',
+        resultStatus: 0, // ACCEPTED
+        autosave: true,
+      },
+    ];
+  };
+
+  const getQueueResultData = (): AutosaveQueueData[] => {
+    return [
+      {
+        applicationReference: 56,
```

Comment:
Can you create an enum for each of the test scenarios you're using just to make it a bit more descriptive instead of just 56, 57, 58 etc

Reply:
Spot on, I'll add this now.